runtime-infra/fragments/ecs-service.yaml

```diff
@@ -1151,6 +1161,108 @@ Resources:
                 Effect: Allow
                 Resource: "*"
 
+  ###                            Event adn Alarm                           ###
+  ############################################################################
+
+  EventRule:
+    Type: AWS::Events::Rule
+    Properties:
+      Name: !Sub "ECSStoppedTasksEvent-${ECSService.Name}"
+      Description: Triggered when an Amazon ECS Task is stopped due to OOM
+      EventPattern:
+        source:
+          - aws.ecs
+        detail-type:
+          - ECS Task State Change
+        detail:
+          desiredStatus:
+            - STOPPED
+          lastStatus:
+            - STOPPED
+          group: 
+            - !Sub "service:${ECSService.Name}"
+          containers:
+            reason:
+              - prefix: "OutOfMemory"
+      State: !Ref MonitorStatus
+      Targets:
+        - Arn: !GetAtt LogGroupOOMEvents.Arn
+          Id: LogTarget
+    
+  # Creating a log group for monitoring the OOM events
+  LogGroupOOMEvents:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/events/ecs-oom-errors-${MicroServiceUniqueName}"
+      RetentionInDays: 14
+      KmsKeyId:
+        Fn::GetAtt:
+          - LogGroupKmsKey
+          - Arn
+
+  # Resource based policy to allow EventBridge to publish to CloudWatch Logs
+  LogGroupOOMEventsPolicy:
+    Type: AWS::Logs::ResourcePolicy
+    Properties:
+      PolicyName: !Sub "EventBridgeToCWLogsPolicy-${ECSService.Name}"
```

Comment:
Use `${MicroServiceUniqueName}` instead of `${ECSService.Name}`

Reply:
done